analysis/README.md

```diff
@@ -4,6 +4,45 @@ This folder contains implementation of Edna data analysis in Python.
 
 ## Usage
 
-First of all, normally you don't want to call this code directly, it is supposed to be called from `edna-server`.
-We use [`poetry`](https://python-poetry.org/) for dependency management, though you can install dependencies using your favorite way and not use `poetry`.
+First, normally you don't want to call this code directly, it is supposed to be called from `edna-server`.
+We use [`poetry`](https://python-poetry.org/) for dependency management, though you can install dependencies using your
+favorite way and not use `poetry`.
 To get shell with all dependencies installed run `poetry install` and then `poetry shell` from this directory.
+
+## Features
+
+The list of features implemented in an analytic module:
+1. Compute parameters of fitted sigmoid.
+2. Detect potential outliers.
+3. Compute parameters of fitted sigmoid without outliers.
+
+### Input format
+
+JSON containing list of experiments in the form:
+```
+[{"experiment": int,
+"data": [[float, float], ...],
```

Comment:
One more thing: AFAIU currently the order of points matters, I have an example where I get different results depending on the order. I think we should state that explicitly. So we have 2 options:
1. Say that the order matters and to get good results the points should be sorted. Then I'll add sorting on my side.
2. Say that the order doesn't matter and update your code to sort points.

I am okay with both options, what's your preference?

Reply:
Can you please share the example? I haven't seen such behavior yet

Reply:
I mean answers can be a bit different but it's a floating point difference. I can sort them explicitly so we always get same results for the matter.

Reply:
Anyway, sorted it)

Reply:
```
> python ic50_analysis.py '[{    "experiment": 2,    "data": [[0.00508, 36144], [0.00508, 37086], [0.01524, 36022], [0.01524, 36787], [0.04572, 36248], [0.04572, 36362], [0.13717, 36385], [0.13717, 36724], [0.41152, 35417], [0.41152, 35992], [1.2346, 35461], [1.2346, 35479], [3.7037, 33434], [3.7037, 33917], [11.111, 27498], [11.111, 27737], [33.333, 16689], [33.333, 17041], [100, 5973], [100, 6045], [150, 1000]]}]'

[{"experiment": 2, "status": "DONE", "params": [36482.46722333145, 1.0438910039400358, 41.646644340626764, -7163.367701146651], "outliers": [20], "new_params": [36404.55941244787, 1.1447795315311728, 33.048994525300735, -2541.319176862586]}]
```

```
> python ic50_analysis.py '[{    "experiment": 2,    "data": [[150, 1000], [0.00508, 36144], [0.00508, 37086], [0.01524, 36022], [0.01524, 36787], [0.04572, 36248], [0.04572, 36362], [0.13717, 36385], [0.13717, 36724], [0.41152, 35417], [0.41152, 35992], [1.2346, 35461], [1.2346, 35479], [3.7037, 33434], [3.7037, 33917], [11.111, 27498], [11.111, 27737], [33.333, 16689], [33.333, 17041], [100, 5973], [100, 6045]]}]'             

/home/gromak/Work/biocad/edna/analysis/ic50/__init__.py:10: RuntimeWarning: overflow encountered in power
  return ((a - d) / (1.0 + ((x / c) ** b))) + d
/home/gromak/.cache/pypoetry/virtualenvs/edna-3Rv1oKBx-py3.9/lib/python3.9/site-packages/scipy/optimize/minpack.py:828: OptimizeWarning: Covariance of the parameters could not be estimated
  warnings.warn('Covariance of the parameters could not be estimated',
/home/gromak/Work/biocad/edna/analysis/ic50/__init__.py:10: RuntimeWarning: overflow encountered in power
  return ((a - d) / (1.0 + ((x / c) ** b))) + d
/home/gromak/Work/biocad/edna/analysis/ic50/__init__.py:10: RuntimeWarning: overflow encountered in power
  return ((a - d) / (1.0 + ((x / c) ** b))) + d
/home/gromak/.cache/pypoetry/virtualenvs/edna-3Rv1oKBx-py3.9/lib/python3.9/site-packages/scipy/optimize/minpack.py:828: OptimizeWarning: Covariance of the parameters could not be estimated
  warnings.warn('Covariance of the parameters could not be estimated',
/home/gromak/.cache/pypoetry/virtualenvs/edna-3Rv1oKBx-py3.9/lib/python3.9/site-packages/scipy/optimize/minpack.py:828: OptimizeWarning: Covariance of the parameters could not be estimated
  warnings.warn('Covariance of the parameters could not be estimated',
[{"experiment": 2, "status": "DONE", "params": [36482.46596376745, 1.0438931709031831, 41.6464311832262, -7163.2768968641385], "outliers": [0, 13, 17, 18], "new_params": [36378.55117971187, 1.2019003263313273, 30.035267718626212, -1148.5322663025368]}]
```